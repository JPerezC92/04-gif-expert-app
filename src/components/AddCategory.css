.form {
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-end;
  min-width: 20em;
  width: 50%;
}

.form__inputValue {
  color: rgb(35, 185, 115);
}

.form__input {
  color: gray;
  max-width: 20em;
  min-width: 15em;
  width: 100%;
}
