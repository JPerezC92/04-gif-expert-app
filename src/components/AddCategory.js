import React, { useState } from "react";
import PropTypes from "prop-types";
import "./AddCategory.css";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((state) => [...state, inputValue]);
      setInputValue("");
    }
  };

  return (
    <form className="form" onSubmit={handleSubmit}>
      <p className="form__inputValue">{inputValue}</p>
      &nbsp;
      <input
        className="form__input"
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        placeholder='Exp. "cat"'
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
