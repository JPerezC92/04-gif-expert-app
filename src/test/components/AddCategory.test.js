import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Pruebas en el archivo <AddCategory />", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("Debe de comparar correctamente con la imagen", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de cambiar el value de la caja de texto ", () => {
    const input = wrapper.find("input");
    const value = "Hola mundo";
    input.simulate("change", { target: { value } });
    const inputValue = wrapper.find("p").text().trim();

    expect(inputValue).toBe(value);
  });

  test("No debe de postear la informacion con submit", () => {
    const form = wrapper.find("form");
    form.simulate("submit", { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe de llamar el setCategories y limpiar la caja de texto", () => {
    const value = "cat";

    // simular change
    const input = wrapper.find("input");
    input.simulate("change", { target: { value } });
    const inputValue = wrapper.find("input").prop("value").trim();
    // simular submit
    const form = wrapper.find("form");
    form.simulate("submit", { preventDefault() {} });
    const cleanInputValue = wrapper.find("input").prop("value").trim();

    expect(inputValue).toBe(value);

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    expect(cleanInputValue).toBe("");
  });
});
