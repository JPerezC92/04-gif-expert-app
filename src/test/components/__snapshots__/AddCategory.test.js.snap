// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo <AddCategory /> Debe de comparar correctamente con la imagen 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <p
    className="form__inputValue"
  />
  Â 
  <input
    className="form__input"
    onChange={[Function]}
    placeholder="Exp. \\"cat\\""
    type="text"
    value=""
  />
</form>
`;
